python/tritonserver/_api/_server.py

```diff
@@ -0,0 +1,1086 @@
+            ]
+            start_time = time.time()
+            while not self._model_unloaded(model_versions) and (
+                (timeout is None) or (time.time() - start_time < timeout)
+            ):
+                time.sleep(polling_interval)
+
+    def metrics(self, metric_format: MetricFormat = MetricFormat.PROMETHEUS) -> str:
+        """Return server and custom metrics
+
+        See c:func:`TRITONSERVER_ServerMetrics()`
+
+        Parameters
+        ----------
+        metric_format : MetricFormat
+            format for metrics
+
+        Returns
+        -------
+        str
+            string containing metrics in specified format
+
+        Examples
+        --------
+        Todo
+        """
+
+        return self._server.metrics().formatted(metric_format)
+
+    class _UnstartedServer(object):
+        def __init__(self):
+            pass
+
+        def __getattribute__(self, name):
+            raise InvalidArgumentError("Server not started")
+
+        def __setattr__(self, name, value):
+            raise InvalidArgumentError("Server not started")
+
+    def _model_unloaded(self, model_versions: list[tuple[str, int]]) -> bool:
+        model_states = self.models()
+        for model_version in model_versions:
+            if model_states[model_version].state not in Server._UNLOADED_STATES:
+                return False
+        return True
+
+    def _model_index(self, exclude_not_ready=False) -> list[Model]:
+        if isinstance(self._server, Server._UnstartedServer):
+            raise InvalidArgumentError("Server not started")
+
+        models = json.loads(
+            self._server.model_index(exclude_not_ready).serialize_to_json()
+        )
+
+        for model in models:
+            if "version" in model:
+                model["version"] = int(model["version"])
+
+        return [Model(self._server, **model) for model in models]
+
+    _UNLOADED_STATES = [None, "UNAVAILABLE"]
+
+
+class Metric(TRITONSERVER_Metric):
+    """Class for adding a custom metric to Triton inference server metrics reporting
+
+    Metric objects are created as part of a MetricFamily where the
+    MetricFamily defines the type and name of the metric. When a
+    metric is added to a family it can further add additional labels
+    allowing for multiple metrics to be associated with the same
+    family. For more details see c:func:`TRITONSERVER_Metric`
+    documentation.
+
+    Examples
+    --------
+    Todo
+
+    """
+
+    def __init__(self, family: MetricFamily, labels: Optional[dict[str, str]] = None):
+        """Initialize Metric object
+
+        Parameters
+        ----------
+        family : MetricFamily
+            Metric family that includes MetricKind, name, and
+            description.
+        labels : Optional[dict[str, str]]
+            Additional labels for the metric (returned in reporting to
+            distinguish multiple metrics within a family)
+
+        Examples
+        --------
+
+        Todo
```

Comment:
Todo

Reply:
will do

Reply:
Just realized there's about ~8 of these "Todo" examples - just mentioning here rather than commenting on each one. I'd be fine with removing the block and adding the example in a follow-up, but I don't think the rendered doc should have the "Todo"s in the rendered examples.